File: find a corresponding node of a binary tree ina clone of that tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

class Solution {
public:
void dfs(TreeNode* root,vector<TreeNode*>&v){
    if(root==nullptr){return ;}
    dfs(root->left,v);
    v.push_back(root);
    dfs(root->right,v);
    return;
}
    TreeNode* getTargetCopy(TreeNode* r1, TreeNode* r2, TreeNode* tar) {
       vector<TreeNode*>v;
       dfs(r2,v);
       for(TreeNode* tmp:v){
           if(tmp->val==tar->val){return tmp;}
       }
       return nullptr;
    }
};

File: find a corresponding node of a binary tree ina clone of that tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */

class Solution {
    void dfs(TreeNode root,ArrayList<TreeNode>arr){
        if(root==null){return;}
        dfs(root.left,arr);
        arr.add(root);
        dfs(root.right,arr);
    }
    public final TreeNode getTargetCopy(final TreeNode original, final TreeNode cloned, final TreeNode target) {
        ArrayList<TreeNode>arr=new ArrayList<TreeNode>();
        dfs(cloned,arr);
        for(TreeNode tmp:arr){
            if(tmp.val==target.val){return tmp;}
        }
        return null;
    }
}